interp/src/primitives/stateful.rs

```diff
@@ -398,6 +291,26 @@ impl StdReg {
 }
 
 impl Primitive for StdReg {
+    fn do_tick(&mut self) -> Vec<(ir::Id, OutputValue)> {
+        //first commit any updates
+        if let Some(val) = self.update.take() {
+            self.data[0] = val;
+        }
+        //then, based on write_en, return
+        if self.write_en {
+            self.write_en = false; //we are done for this cycle
+                                   //if do_tick() is called again w/o an execute() preceeding it,
+                                   //then done will be low.
+            vec![
+                (ir::Id::from("out"), self.data[0].clone().into()),
+                (ir::Id::from("done"), Value::bit_high().into()),
+            ]
+        } else {
+            //done is low, but there is still data in this reg to return
+            vec![(ir::Id::from("out"), self.data[0].clone().into())]
```

Comment:
This should return a low `done` since it means that the register was ticked w/o a high write enable and it doesn't seem like you're setting the done low anywhere else (which I think is correct since this is the spot that it would make the most sense to do it)

Reply:
I remember I initially did return a low `done` and (maybe coincidentally) the tests became infinite-loops. I'm pretty sure the `done` ports are set low based on the zero-assign functionality of `interpret-group`? I'll try adding it again and see what happens.

Reply:
Done is an output port so zero-assigns shouldn't mess with it

Reply:
hm alright, setting the `done` ports low does result in an infinite loop which probably stems from `zero-assigns`, I'll try to sort it out